src/Particle/CMakeLists.txt

```diff
@@ -53,6 +51,34 @@ target_include_directories(qmcparticle PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
 target_link_libraries(qmcparticle PRIVATE platform_cpu_LA)
 target_link_libraries(qmcparticle PUBLIC qmcnumerics qmcutil platform_runtime)
 
+# Workaround for broken OMP Target debug in clang as of 5d2ce7663b10c107328a4ae0c678165209e64619
+# but clang 15 main is completely broken for our omptarget code,
+#
+# In the meantime we need to make a new target for just the omp target tainted files
+# so we don't have broken debug for all of ParticleSet.
+
+# this can be put back into qmcparticle when omp target isn't broken.
+set(PARTICLEOMPTARGET
+    createDistanceTableAAOMPTarget.cpp
+    createDistanceTableABOMPTarget.cpp)
+
+if(USE_OBJECT_TARGET)
+  add_library(qmcparticle_omptarget OBJECT ${PARTICLEOMPTARGET})
+else(USE_OBJECT_TARGET)
+  add_library(qmcparticle_omptarget ${PARTICLEOMPTARGET})
+endif(USE_OBJECT_TARGET)
+
+target_include_directories(qmcparticle_omptarget PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+target_link_libraries(qmcparticle_omptarget PUBLIC qmcutil containers)
+
+if(ENABLE_OFFLOAD AND ${COMPILER} MATCHES "Clang")
+    target_compile_options(qmcparticle_omptarget PRIVATE $<$<CONFIG:DEBUG>:-O3>)
+endif()
```

Comment:
Could you introduce a CMake option `ENABLE_OFFLOAD_CLANG_DEBUG_O3` at top level CMakeLists.txt and repetitive comments next to this option.